extensions/panache/panache-common/deployment/src/main/java/io/quarkus/panache/common/deployment/PanacheEntityEnhancer.java

```diff
@@ -91,10 +91,19 @@ public PanacheEntityClassVisitor(String className, ClassVisitor outputClassVisit
 
         @Override
         public FieldVisitor visitField(int access, String name, String descriptor, String signature, Object value) {
-            FieldVisitor superVisitor = super.visitField(access, name, descriptor, signature, value);
             EntityField ef = fields.get(name);
             if (ef == null) {
-                return superVisitor;
+                return super.visitField(access, name, descriptor, signature, value);
+            }
+            //we make the fields protected
```

Comment:
Why `protected` and not `private`?
Note that this may impact the jsonb/jackson/jaxb marshaller strategy, but probably tests would expose this.

Reply:
Because I think we allow sub classes to access parent fields directly.

We can remove this from the PR, but I thought it would be interesting to add, as I assumed that nothing would be using the fields directly but it looks like this is not the case.

Reply:
Mmm, I think we don't actually allow field access from within the entity, except in the constructor.

Reply:
panache rest data uses the id field, it was originally private but it failed because of this so I changed it. Maybe I should move this into a seperate pr? It's not really needed for this one, I thought it would just work and make these sort of errors more obvious.